rust_snuba/rust_arroyo/src/processing/mod.rs

```diff
@@ -130,14 +130,23 @@ impl<'a, TPayload: 'static + Clone> StreamProcessor<'a, TPayload> {
     pub fn subscribe(&mut self, topic: Topic) {
         let callbacks: Box<dyn AssignmentCallbacks> =
             Box::new(Callbacks::new(self.strategies.clone()));
-        self.consumer.subscribe(&[topic], callbacks).unwrap();
+        self.consumer
+            .lock()
+            .unwrap()
+            .subscribe(&[topic], callbacks)
+            .unwrap();
     }
 
     pub fn run_once(&mut self) -> Result<(), RunError> {
         if self.is_paused {
             // If the consumer waas paused, it should not be returning any messages
             // on ``poll``.
-            let res = self.consumer.poll(Some(Duration::ZERO)).unwrap();
+            let res = self
+                .consumer
+                .lock()
+                .unwrap()
+                .poll(Some(Duration::ZERO))
+                .unwrap();
```

Comment:
Isn't unwrap here going to panic in case of errors instead of surfacing the same error in the `Result` of the `run_once` method ?

Reply:
Will make a note to address it later. But not in this PR, this code hasn't changed here.